SKFlatMaker/src/SKFlatMaker.cc

```diff
@@ -2917,15 +2969,59 @@ void SKFlatMaker::fillFatJet(const edm::Event &iEvent)
+
+    if(DataYear==2016){
+      //=== 2016 jet IDs                                                                                                                                                                                    
+      //==== https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVRun2016                                                                                                                                     
+      if(fabs(eta)>2.7){
+	//Note that the jet ID requirements for |eta|>2.7 are not recommended for PUPPI jets (see e.g. https://indico.cern.ch/event/578287/contributions/2347317/).
+
+        tightJetID = false;
+        tightLepVetoJetID = false;
+      }
+      else {
+        tightJetID        = (NHF<0.90 && NEMF<0.90 && NumConst>1) &&            ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.99) || fabs(eta)>2.4) && fabs(eta)<=2.7;
+        tightLepVetoJetID = (NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8) && ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.90) || fabs(eta)>2.4) && fabs(eta)<=2.7;
+      }
     }
-    else if(fabs(eta)>2.7){
-      tightJetID = (NEMF>0.02 && NEMF<0.99) && (NumNeutralParticles>2) && abs(eta)>2.7 && abs(eta)<=3.0;
-      tightLepVetoJetID = false; //FIXME default is false or true? following CAT2016
+    else   if(DataYear<=2017){
+      //=== 2017 jet IDs                                                                                                                                                                                    
+      //==== https://twiki.cern.ch/twiki/bin/view/CMS/JetID13TeVRun2017                                                                                                                                    
+      if(fabs(eta)>3.0){
+        tightJetID = (NEMF<0.90 && NHF>0.02 && NumNeutralParticles> 2 && NumNeutralParticles < 15);
+        tightLepVetoJetID = (NEMF<0.90 && NHF>0.02 && NumNeutralParticles>2 && NumNeutralParticles < 15);
+      }
+      else if(fabs(eta)>2.7){
+        tightJetID = (NEMF<0.99);
+        tightLepVetoJetID =  (NEMF<0.99);
+      }
+      else{
+        tightJetID        = (NHF<0.90 && NEMF<0.90 && NumConst>1) &&            ((fabs(eta)<=2.4 && CHF>0 && CHM>0)              || fabs(eta)>2.4) && fabs(eta)<=2.7;
+        tightLepVetoJetID = (NHF<0.90 && NEMF<0.90 && NumConst>1 && MUF<0.8) && ((fabs(eta)<=2.4 && CHF>0 && CHM>0 && CEMF<0.80) || fabs(eta)>2.4) && fabs(eta)<=2.7;
+      }
+    }
+    else{
+      //=== 2018 jet ID                                                                                                                                                                                     
+      //==== https://twiki.cern.ch/twiki/bin/viewauth/CMS/JetID13TeVRun2018                                                                                                                                 
+      if(fabs(eta)>3.0){
+        tightJetID = (NEMF<0.90 && NHF>0.2 && NumNeutralParticles>10 );
```

Comment:
NHF>0.2 should be NHF>0.02

Reply:
This was actually correct, the table in the twiki just fixed.

Reply:
The original values are correct right?

Reply:
Yep, 0.2 was the correct value.